src/core/server/saved_objects/routes/get.ts

```diff
@@ -7,15 +7,18 @@
  */
 
 import { schema } from '@kbn/config-schema';
-import { IRouter } from '../../http';
 import { InternalCoreUsageDataSetup } from '../../core_usage_data';
+import type { InternalSavedObjectRouter } from '../internal_types';
 import { catchAndReturnBoomErrors } from './utils';
 
 interface RouteDependencies {
   coreUsageData: InternalCoreUsageDataSetup;
 }
 
-export const registerGetRoute = (router: IRouter, { coreUsageData }: RouteDependencies) => {
+export const registerGetRoute = (
+  router: InternalSavedObjectRouter,
+  { coreUsageData }: RouteDependencies
```

Comment:
I suspect this is going to be problematic too down the line (SavedObjects -> CoreUsageData -> SavedObjects) but that's a completely different issue.